modules/core/shared/src/main/scala/Trace.scala

```diff
@@ -140,7 +177,7 @@ object Trace {
 
   }
 
-  implicit def liftKleisli[F[_], E](implicit trace: Trace[F]): Trace[Kleisli[F, E, *]] =
+  implicit def liftKleisli[F[_]: MonadCancelThrow, E](implicit trace: Trace[F]): Trace[Kleisli[F, E, *]] =
```

Comment:
I have a PR futzing with `Resource#mapK` constraints but I don't think it helps much here.
- https://github.com/typelevel/cats-effect/pull/2646

I think I encountered this working on Bayou, and I wondered if it was a good argument to have two `Trace` typeclasses after all.

Reply:
I don't think that PR helps much here: you still need `MonadCancelThrow[F]`, and we're going `F ~> G` and `G ~> F` unless someone is more clever than me.  (Possible!  Nay, probable!)

The base instances, other than `noop`, all already require `MonadCancelThrow`, so I don't think this limits which effects can be traced.  But it could require a stronger constraint in a few places along the way.